### DISCUSSION TITLE
Loop set up (RINEX / sp3 files)

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#132] [Q&A] [answered]
Hello

I am running the “Prepare Leo Data” script for a cubesat mission and have a question with regards to running the loop program and best way to set the script up.

I have converted the RINEX files (see below example) with gfzrnx to daily files with some of them not having a 0000 time portion (i.e. 0100 time stamp) in the file name. So, if I run the loop program with 1 day I will miss the ones that are not 0000.
 In a similar way when I want to convert the sp3 files to GROOPS format with SP3FORMAT2ORBIT I would need to have the loop program do steps of 1 sec to not miss the files that are different from 0000 in time stamp. I can set up the loop in a step of seconds but then I will basically run 86400 loops per day to get about 20 files or so per day as not all the times are covered.

Is there another way to run the loop program to find all the different files without having to do steps in seconds (like a wildcard option)?

Txs for the help

BR
Jean

SP3 file structure
Sat_2021-08-01T00-01-56.sp3
Sat_2021-08-01T02-01-21.sp3
Sat_2021-08-01T03-02-19.sp3
Sat_2021-08-01T05-16-49.sp3
Sat_2021-08-01T07-17-19.sp3
Sat_2021-08-01T08-07-20.sp3
Sat_2021-08-01T09-01-49.sp3
Sat_2021-08-01T10-01-21.sp3
Sat_2021-08-01T11-22-19.sp3
Sat_2021-08-01T13-01-49.sp3
Sat_2021-08-01T14-01-20.sp3

RINEX file structure
SAT00XXX_R_20212130000_01D_10S_GO.rnx
SAT00XXX_R_20212140000_01D_10S_GO.rnx
SAT00XXX_R_20212150000_01D_10S_GO.rnx
SAT00XXX_R_20212160000_01D_10S_GO.rnx
SAT00XXX_R_20212170100_01D_10S_GO.rnx <<---
SAT00XXX_R_20212180000_01D_10S_GO.rnx 
SAT00XXX_R_20212190000_01D_10S_GO.rnx
SAT00XXX_R_20212200000_01D_10S_GO.rnx
SAT00XXX_R_20212210100_01D_10S_GO.rnx <<---

**Comment by tmayerguerr** [comment, marked as answer]
Dear Jean,

Wildcard options are not supported as there is no system independent supporting c++ standard at the moment.

But with a trick it is possible: Add in your temporal LoopProgram another LoopProgram with the loop->commandOutput. This command must return the filename. The Linux command looks like
`ls -1 Sat_{loopTime:%D}T*.sp3`, with Windows like `dir /B Sat_{loopTime:%D}T*.sp3`.

regards
Torsten

**Comment by Bemtgenj** [comment]
Dear Torsten

Txs for the feedback. I will try out your "trick"

Cheers
Jean
